components/security/core/authentication.rst

```diff
@@ -0,0 +1,229 @@
+.. index::
+    single: Security, Core Authentication
+
+Security Core: Authentication
+=============================
+
+As you've :ref:`already learned <components-security-terminology>`, the first
+step in the Security system is authentication, answering the question: **Who
+are you?**
+
+Authentication Manager
+----------------------
+
+This question is answered by an
+:class:`authentication manager <Symfony\\Component\\Security\\Core\\Authentication\\AuthenticationManagerInterface>`.
+This class recieves a token containing the information available about the
+visitor of your application. Based on this information, it tries to determine
+the visitor and returns an authenticated token. If it can't authenticate the
+user, an
+:class:`Symfony\\Component\\Security\\Core\\Exception\\AuthenticationException`
+is thrown. This exception can the be transformed into a redirect to the login
+form, for example.
+
+Using the authentication manager, authenticating a user becomes very simple (no
+worries, you'll fill the missing variables within a couple of minutes)::
+
+    use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+    try {
+        $authenticatedToken = $authManager->authenticate($unauthenticatedToken);
+    } catch (AuthenticationException $e) {
+        // ... authentication failed, do something
+    }
+
+The Security component comes with one very flexible authentication manager, the
+:class:`Symfony\\Component\\Security\\Core\\Authentication\\AuthenticationProviderManager`.
+This uses authentication providers to authenticate a token.
+
+.. _component-security-core-authentication-providers:
+
+Authentication Providers
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+One ``AuthenticationProviderManager`` can have multiple
+:class:`authentication providers <Symfony\\Component\\Security\\Core\\Authentication\\Provider\\AuthenticationProviderInterface>`.
+Using the ``supports()`` method, providers determines if they can authenticate
+the passed token.
+
+The most simple authentication provider is the
```

Comment:
This paragraph needs some additional explanation about how can this auth provider possibly exist. Why do we need to authenticate anonymous users?